### PR TITLE
Feature/ui static

### PR DESCRIPTION
**Pull request by LostFan**
Review, please

### CODE REVIEW

File: src/main/webapp/js/ideas/ideas.factory.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+        var publicMethod = {
+            getIdeas: getIdeas,
+            updateIdea: updateIdea
+        };
+        return publicMethod;
+
+        function getIdeas() {
+            var ideas = crudFactory.ideas().get().$promise;
+            return ideas;
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/js/ideas/ideas.factory.js]: may be, it makes sense to rename local variable 'ideas' into 'promise'

File: src/main/webapp/js/crud.factory.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+
+(function () {
+    'use strict';
+    angular
+        .module('ideaFactories')
+        .factory('crudFactory', crudFactory);
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/js/crud.factory.js]: the name 'crudFactory' seems to be not enough sensible

File: src/main/webapp/js/ideas/ideas.factory.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        };
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/js/ideas/ideas.factory.js:21]: maybe it makes sense to rename variable 'ideas' into 'promise' (or remove it)

File: src/main/webapp/js/services.js [hunk 4]
```diff
@@ -2,16 +2,20 @@
...
        return {
         getIdeas:function (tag) {
-          var newData = Data.getIdeas();
+         var newData;
+         ideasFactory.getIdeas().then(function (ideas) {
+            newData = ideas;
+         });
+
           if(tag !== undefined) {
               newData = [];
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/js/services.js]: This method should be refactored to call factory method only once to to decrease number of server calls.

File: src/main/webapp/js/services.js [hunk 5]
```diff
@@ -2,16 +2,20 @@
...
-              for(var i=0;i<Data.getIdeas().length; i++)
-                if(tag.toLowerCase() ==Data.getIdeas()[i].tag.toLowerCase())
-                  newData.push(Data.getIdeas()[i]);
+              for(var i=0;i<ideasFactory.getIdeas().length; i++)
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/js/services.js]: This method should be refactored to call factory method only once to decrease number of server calls.

File: src/main/webapp/js/tags/tags.factory.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        var publicMethod = {
+            getTopNTags: getTopNTags,
+            getIdeasByTag: getIdeasByTag
+        };
+        return publicMethod;
+
+        function getTopNTags() {
+            var tags = crudFactory.tags().get().$promise;
+            return tags;
+        }
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/js/tags/tags.factory.js]: maybe it worth to rename the local variable 'tags' to 'promise'

File: src/main/webapp/js/tags/tags.factory.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+        var publicMethod = {
+            getTopNTags: getTopNTags,
+            getIdeasByTag: getIdeasByTag
+        };
+        return publicMethod;
+
+        function getTopNTags() {
+            var tags = crudFactory.tags().get().$promise;
+            return tags;
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/js/tags/tags.factory.js]: maybe it worth to rename the local variable 'tags' to 'promise'

File: src/main/webapp/js/tags/tags.factory.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+
+      function getIdeasByTag(tag) {
+        if( tag === undefined)
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/js/tags/tags.factory.js:26]: variable 'ideas' should be declared only once (and be renamed) for clarity

**Comment by Hanna-Aliakseichykava** [comment]
:+1:

**Comment by reshetnev** [comment]
:+1:

**Pull request merged**